weld-python/weld/lazy.py

```diff
@@ -301,3 +302,83 @@ def identity(phys_value, decoder):
     Creates an identity `WeldLazy` from a `PhysicalValue` and decoder.
     """
     return WeldLazy(str(phys_value.id), [phys_value], phys_value.output_type, decoder)
+
+class WeldFunc(object):
+    """
+    The result of a function annotated with `welfunc`.
+
+    This object is effectively a function with a field `code`, which extracts the raw code
+    constructed by the annotated function without constructing a `WeldLazy`.
```

Comment:
Can you add a comment that `WeldFunc` object shouldn't be consumed by anything? Rather, it's expected that the `__callable__` function is called on this object to get a `WeldLazy` object that can be used downstream.

Reply:
Hmm, can you clarify? The API for this doesn't prevent a caller from accessing code or doing something else with this (though it's not very useful for anything other than calling)

Reply:
Feel free to ignore this. Just wanted a clarifying comment saying dependencies are tracked using `WeldLazy`, and you need to call the `__callable__` function to get this.